File: server/internal/auth/domain/schemas.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// User represents an entity with personal and account-related data managed within the system.
type User struct {
	ID           uuid.UUID
	Email        string
	PasswordHash string
	FullName     string

	CreatedAt   time.Time
	UpdatedAt   time.Time
	ActivatedAt *time.Time
	ArchivedAt  *time.Time
}

type Permission struct {
	ID          string    `json:"id"`
	Module      string    `json:"module"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
}

type Role struct {
	ID   int    `json:"id"`
	Name string `json:"name"`
}

type AuthTokens struct {
	AccessToken      string    `json:"access_token"`
	RefreshToken     string    `json:"refresh_token"`
	AccessExpiresAt  time.Time `json:"access_expires_at"`
	RefreshExpiresAt time.Time `json:"refresh_expires_at"`
}

type Session struct {
	ID               uuid.UUID
	UserID           uuid.UUID
	RefreshTokenHash string
	ExpiresAt        time.Time
	CreatedAt        time.Time
	UpdatedAt        time.Time
	RevokedAt        *time.Time
}

File: client/libs/api/src/lib/features/auth/types/schemas.ts
export interface AuthSession {
    accessToken: string;
    accessExpiresAt: string;
    refreshToken?: string;
    refreshExpiresAt?: string;
}

export interface Role {
    id: number;
    name: string;
}

export interface User {
    id: string;
    email: string;
    full_name: string;
    roles?: Role[];
    created_at?: string;
    updated_at?: string;
}

export type Menu = {
    id: string;
    // domain: string;
    label: string;
    path?: string;
    icon?: string;
    // order?: number;
    // parentId?: string;
    // permissions?: string[];
    // visible: boolean;
    children?: Menu[];
};
